hydromt/models/components/grid.py

```diff
@@ -13,371 +11,165 @@
-            f"on {reclass_table_fn}"
-        )
-        # Read raster data and remapping table
-        da = self.data_catalog.get_rasterdataset(
-            raster_fn, geom=self.region, buffer=2, variables=variable, **kwargs
-        )
-        if not isinstance(da, xr.DataArray):
-            raise ValueError(
-                f"raster_fn {raster_fn} should be a single variable. "
-                "Please select one using the 'variable' argument"
+            Additional keyword arguments to be passed to the `write_nc` method.
+        gdal_compliant : bool, optional
+            If True, write grid data in a way that is compatible with GDAL,
+            by default False
+        rename_dims: bool, optional
+            If True and gdal_compliant, rename x_dim and y_dim to standard names
+            depending on the CRS (x/y for projected and lat/lon for geographic).
+        force_sn: bool, optional
+            If True and gdal_compliant, forces the dataset to have
+            South -> North orientation.
+        """
+        if len(self.data) == 0:
+            _exec_nodata_strat(
+                msg="No grid data found, skip writing.",
+                strategy=NoDataStrategy.IGNORE,
+                logger=logger,
+            )
+        else:
+            if not self.root.is_writing_mode():
+                raise IOError("Model opened in read-only mode")
+            # write_nc requires dict - use dummy 'grid' key
+            return write_nc(  # Can return DeferedFileClose object
+                {"grid": self._data},
+                fn,
+                gdal_compliant=gdal_compliant,
+                rename_dims=rename_dims,
+                force_sn=force_sn,
+                **kwargs,
             )
-        df_vars = self.data_catalog.get_dataframe(
-            reclass_table_fn, variables=reclass_variables
-        )
-        # Data resampling
-        ds_vars = workflows.grid.grid_from_raster_reclass(
-            grid_like=self.grid,
-            da=da,
-            reclass_table=df_vars,
-            reclass_variables=reclass_variables,
-            fill_method=fill_method,
-            reproject_method=reproject_method,
-            mask_name=mask_name,
-            rename=rename,
-        )
-        # Add to maps
-        self.set_grid(ds_vars.rename(rename))
-
-        return list(ds_vars.data_vars.keys())
 
-    def setup_grid_from_geodataframe(
+    def read(
         self,
-        vector_fn: Union[str, Path, gpd.GeoDataFrame],
-        variables: Optional[Union[List, str]] = None,
-        nodata: Optional[Union[List, int, float]] = -1,
-        rasterize_method: Optional[str] = "value",
-        mask_name: Optional[str] = "mask",
-        rename: Optional[Dict] = None,
-        all_touched: Optional[bool] = True,
-    ) -> List[str]:
-        """HYDROMT CORE METHOD: Add data variable(s) to grid object by rasterizing the data from ``vector_fn``.
+        fn: str = "grid/grid.nc",
```

Comment:
definitely something that we should have a larger discussion about but I'm really not fond of this abbreviation, not least of which because it's not a filename and it can also be many things besides a path, such as the actual data or a catalog entry. (at least it could be in the original implementation) Can we at least pick something more descriptive like "rel_path" in the case of string/path or "data_like" or anything that hints at all the things it could be?

Reply:
I agree. I changed it data_like for now, but maybe good to discuss this and decide if we need to have a standardized name across the code base for these arguments.